### ISSUE TITLE
Unable to download semcov v1.16.0 dependency

### ISSUE THREAD

**Issue by pmoncadaisla** [open-telemetry/opentelemetry-go#3598]
### Description

Unable to download semcov v1.16.0 dependency.
I have tried to add the example code in our monorepo https://github.com/open-telemetry/opentelemetry-go/tree/v1.11.2/example/otel-collector. Our monorepo has a single `go.mod`.
We use `vendored` dependencies (`go mod tidy` && `go mod vendor`)

It seems like `v1.16.0` is available at https://github.com/open-telemetry/opentelemetry-go/tree/main/semconv/v1.16.0 but that version is not listed in https://pkg.go.dev/go.opentelemetry.io/otel/semconv, the latest version listed there is `v1.12.0`.

### Environment

- OS: macOS
- Architecture: darwin/arm64
- Go Version: go1.19.2
- opentelemetry-go version: v1.11.2

### Steps To Reproduce

Create a `main.go` with the content described in https://github.com/open-telemetry/opentelemetry-go/tree/v1.11.2/example/otel-collector

When executing `go mod tidy` it can't find the package `go.opentelemetry.io/otel/semconv/v1.16.0`

```
❯ go mod tidy
go: finding module for package go.opentelemetry.io/otel/semconv/v1.16.0
github.com/masmovil/mm-monorepo/pkg/mas-stack/bss/orgs/msisdn-inventory-api/pkg/tracing imports
        go.opentelemetry.io/otel/semconv/v1.16.0: module go.opentelemetry.io/otel@latest found (v1.11.2), but does not contain package go.opentelemetry.io/otel/semconv/v1.16.0
 ``` 

### Expected behavior

Execute `go mod tidy` && `go mod vendor` successfuly.
It seems like it is expecting to find the `v1.16.0` version listed in https://pkg.go.dev/go.opentelemetry.io/otel/semconv

**Comment by MrAlias** [comment]
The "go.opentelemetry.io/otel/semconv/v1.16.0" package has not been released yet, it is not included in the `v1.11.2` release. It will be included in the next release.

**Comment by pmoncadaisla** [comment]
I see!

My mistake, I checked the example in the `master` branch rather that the latest released tag.
Thanks!
